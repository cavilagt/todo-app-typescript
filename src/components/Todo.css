.actions {
  display: flex;
  align-items: center;
}

.todo {
  display: flex;
  cursor: pointer;
}

.todo .content {
    flex-grow: 2;
}

.done {
  opacity: .7;
}

.done:hover {
  opacity: 1;
}

.todo .icon:hover {
  animation: blinker 1.5s linear infinite;
  cursor: pointer;
}

@keyframes blinker {
  30% {
    opacity: 0;
  }
}