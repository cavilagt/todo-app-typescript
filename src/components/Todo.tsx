import React, { useContext } from "react";
import { TodoType, TodoStatus } from "../types/State";

import "./Todo.css";
import AppContext from "../context/AppContext";

type ComponentProps =  TodoType;

const Todo:React.FunctionComponent< ComponentProps > = ({ title, status, uid, description }) => {

  const { handlerViewDetails, handlerDeleteTodo, handlerToggleTodo} = useContext(AppContext);

  const onClickDelete = () => handlerDeleteTodo && handlerDeleteTodo( uid );
  const onToggleTodo = () => handlerToggleTodo && handlerToggleTodo( uid );
  const onDobleClickTodo = () => handlerViewDetails && handlerViewDetails( uid );

  const completed = status === TodoStatus.COMPLETED;

  const statusClass = completed? 'teal inverted todo done': 'todo';

  return (
    <div className={`ui segment ${ statusClass }`} onDoubleClick={ onDobleClickTodo }>
      <div className="content">
        <label className="ui header" >{title}</label>
        <p>{description}</p>
      </div>
      <div className="actions">
        <i className={`circle ${completed? 'check': ''} icon big`} onClick={ onToggleTodo }></i>
        <i className="close icon big" onClick={ onClickDelete }></i>
      </div>
    </div>
  );
};

export default Todo;
