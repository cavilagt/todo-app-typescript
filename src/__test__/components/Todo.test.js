import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Todo from "../../components/Todo";

describe("<Todo /> component", ()=> {
  
  afterEach(cleanup);
  
  it("renders without crashing", ()=>{
    const rootDiv = document.createElement("div");
    ReactDOM.render( <Todo />, rootDiv );
  });

  it("renders correctly, completed task", ()=>{
    const rootDiv = document.createElement("div");
    const todo = {title:'Title todo test', status:"completed", uid:"1234", description:"A description, just for testing"};
    const { getByTestId } = render( <Todo {...todo} />, rootDiv );

    expect( getByTestId( todo.uid )).toMatchSnapshot();
  });

  it("renders correctly, completed task", ()=>{
    const rootDiv = document.createElement("div");
    const todo = {title:'Incompleted task', status:"incompleted", uid:"4567", description:"A description, just for testing"};
    const { getByTestId } = render( <Todo {...todo} />, rootDiv );

    expect( getByTestId( todo.uid )).toMatchSnapshot();
  });
});