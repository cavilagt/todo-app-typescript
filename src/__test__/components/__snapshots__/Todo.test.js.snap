// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> component renders correctly, completed task 1`] = `
<div
  class="ui segment teal inverted todo done"
  data-testid="1234"
>
  <div
    class="content"
  >
    <label
      class="ui header"
    >
      Title todo test
    </label>
    <p>
      A description, just for testing
    </p>
  </div>
  <div
    class="actions"
  >
    <span
      data-tooltip="Mark as incomplete"
    >
      <i
        class="circle check icon big"
      />
    </span>
    <span
      data-tooltip="Details"
    >
      <i
        class="info circle icon big"
      />
    </span>
    <span
      data-tooltip="Delete"
    >
      <i
        class="times circle icon big"
      />
    </span>
  </div>
</div>
`;

exports[`<Todo /> component renders correctly, completed task 2`] = `
<div
  class="ui segment todo"
  data-testid="4567"
>
  <div
    class="content"
  >
    <label
      class="ui header"
    >
      Incompleted task
    </label>
    <p>
      A description, just for testing
    </p>
  </div>
  <div
    class="actions"
  >
    <span
      data-tooltip="Mark as complete"
    >
      <i
        class="circle icon big"
      />
    </span>
    <span
      data-tooltip="Details"
    >
      <i
        class="info circle icon big"
      />
    </span>
    <span
      data-tooltip="Delete"
    >
      <i
        class="times circle icon big"
      />
    </span>
  </div>
</div>
`;
